gammapy/scripts/analysis.py

```diff
@@ -81,15 +85,21 @@ def settings(self):
         """Configuration settings for the analysis session."""
         return self.config.settings
 
-    def fit(self, optimize_opts=None):
+    def run_fit(self, optimize_opts=None):
         """Fitting reduced data sets to model."""
-        if self.settings["reduction"]["data_reducer"] == "1d":
-            if self._validate_fitting_settings_1d():
-                self._read_model()
-                self._fit_reduced_data_1d(optimize_opts=optimize_opts)
-        else:
-            # TODO raise error?
-            log.info("Fitting available only for joint likelihood with 1D spectrum.")
+        if self._validate_fitting_settings():
+            for obs in self.datasets.datasets:
+                # TODO: fit_range handled in jsonschema validation class
+                if "fit_range" in self.settings["fit"]:
+                    e_min = u.Quantity(self.settings["fit"]["fit_range"]["min"])
+                    e_max = u.Quantity(self.settings["fit"]["fit_range"]["max"])
+                    obs.mask_fit = obs.counts.energy_mask(e_min, e_max)
+                obs.model = self.model
+            log.info("Fitting with joint likelihood.")
```

Comment:
Nitpicky comment: currently for 3D it's not a joint-likelihood fit, but we stack the data before.